app/controllers/follows_controller.rb

```diff
@@ -30,18 +32,25 @@ def create
                  else
                    User.find(params[:followable_id])
                  end
+
     need_notification = Follow.need_new_follower_notification_for?(followable.class.name)
+
     @result = if params[:verb] == "unfollow"
-                follow = current_user.stop_following(followable)
-                Notification.send_new_follower_notification_without_delay(follow, true) if need_notification
-                "unfollowed"
+                unfollow(followable, need_notification: need_notification)
               else
-                follow = current_user.follow(followable)
-                Notification.send_new_follower_notification(follow) if need_notification
-                "followed"
+                rate_limiter = RateLimitChecker.new(current_user)
+                raise RateLimitChecker::DailyFollowAccountLimitReached if rate_limiter.limit_by_action("follow_account")
```

Comment:
Here raising and catching an exception is used for a control flow which is not the best practice because it's harder to read and exceptions are relatively slow in ruby. What do you think about implementing this logic without raising an exception?

Reply:
I honestly did it this way because that's how I saw it being used with rate limiting image uploads and I wanted to follow the style this codebase already uses, but I agree that it might be more readable to do it without an exception. Should I do that?